### PR TITLE
feat(cli,world): deploy writes to worlds.json

### PR DESCRIPTION
**Pull request by holic**
related to #680 

Puts all the latest world addresses by chain into a `worlds.json` file (by default, but is configurable) so that we can have one file to import/read from rather than an import per chain.

This intentionally doesn't include the block number for local chains so the file can remain somewhat consistent between local deploys, since the file is designed to be checked in. We'll lean on [the new `emit HelloWorld()`](#742) to find the block number by querying for the event at the start of the sync.

### CODE REVIEW

**Comment by ludns** [comment]
The issue with the event approach is I am not sure RPCs will let you query for the event over the whole chain; or at the very least you will need to partition that into multiple requests (and have some code to determine from the RPC what an acceptable event range is).

Checking for an event still requires you to go through the bloom filter for every block. It's a linear operation.

We could check for an Erigon node, and then use [`ots_getContractCreator` ](https://github.com/otterscan/otterscan/blob/develop/docs/custom-jsonrpc.md) to find the TX where the contract got deployed.

**Comment by holic** [comment]
> Checking for an event still requires you to go through the bloom filter for every block. It's a linear operation.

TIL!

**Comment by holic** [comment]
> The issue with the event approach is I am not sure RPCs will let you query for the event over the whole chain; or at the very least you will need to partition that into multiple requests (and have some code to determine from the RPC what an acceptable event range is).

Alchemy let's me:

image

(for event `hello()` from OKPC contract deployed at block 14434985)

**Comment by holic** [comment]
> Hopefully Anvil can do it too :)
Anvil has no trouble with it either!

**Comment by holic** [comment]
gonna wait on this until v2 templates land, then carry over the client changes there

File: examples/minimal/packages/contracts/worlds.json
Comment (holic) [inline examples/minimal/packages/contracts/worlds.json:8]: another potential structure for this is

```
[{
  chainId: 31337,
  address: "0x...",
  blockNumber: 123456,
}],
```
but I figured that'd be a little more painful to "splice" if we need to replace a world address/block number for a particular chain

File: templates/vanilla/packages/contracts/deploys/31337/latest.json
Comment (alvrs) [inline templates/vanilla/packages/contracts/deploys/31337/latest.json:3]: should we git ignore this file?

**Review by alvrs — APPROVED** [review]

File: templates/vanilla/packages/contracts/deploys/31337/latest.json
Comment (holic) [reply templates/vanilla/packages/contracts/deploys/31337/latest.json]: whoops, it was ignored but forgot to delete

**Pull request merged**
